File: golang/user_link.go
package main

import (
	"log"

	"github.com/line/line-bot-sdk-go/v7/linebot"
)

func main() {
	bot, err := linebot.New(
		"LINE_CHANNEL_SECRET", // 上書きする
		"LINE_CHANNEL_TOKEN",  // 上書きする
	)
	if err != nil {
		log.Fatal(err)
	}

	// === リッチメニューの情報を見る ===
	if res, _ := bot.GetRichMenuList().Do(); res != nil {
		for _, richmenu := range res {
			log.Printf("%v\n", richmenu)
		}
	}

	// userId := "Uxxxxxx"
	// richmenuId := "richmenu-xxx"

	// // === 個別リッチメニューリンク ===
	// if res, _ := bot.LinkUserRichMenu(userId, richmenuId).Do(); res != nil {
	// 	log.Printf("Response %v;", res)
	// }

	// if res, _ := bot.GetUserRichMenu(userId).Do(); res != nil {
	// 	log.Printf("Response %v;", res)
	// }

	// if res, _ := bot.UnlinkUserRichMenu(userId).Do(); res != nil {
	// 	log.Printf("Response %v;", res)
	// }

	// // === 一括リッチメニューリンク ===
	// if res, _ := bot.BulkLinkRichMenu(richmenuId, userId, userId).Do(); res != nil {
	// 	log.Printf("Response %v;", res)
	// }

	// if res, _ := bot.GetUserRichMenu(userId).Do(); res != nil {
	// 	log.Printf("Response %v;", res)
	// }

	// if res, _ := bot.BulkUnlinkRichMenu(userId, userId).Do(); res != nil {
	// 	log.Printf("Response %v;", res)
	// }
}

File: ruby/user_link.rb
require 'line/bot'
require 'json'

def client
    @client ||= Line::Bot::Client.new { |config|
      config.channel_secret = "LINE_CHANNEL_SECRET" # 上書きする
      config.channel_token = "LINE_CHANNEL_TOKEN"   # 上書きする
    }
end

# === リッチメニューの情報を見る ===
response = client.get_rich_menus
richmenu_list = JSON.parse(response.body).dig('richmenus')
richmenu_list.each do |richmenu|
  p richmenu['richMenuId'], richmenu
end
user_id = 'Uxxxxxx'
richmenu_id = 'richmenu-xxx'

# # === 個別リッチメニューリンク ===
# client.link_user_rich_menu(user_id, richmenu_id)
# p client.get_user_rich_menu(user_id)
# client.unlink_user_rich_menu(user_id)

# # === 一括リッチメニューリンク ===
# client.bulk_link_rich_menus([user_id], richmenu_id)
# p client.get_user_rich_menu(user_id)
# client.bulk_unlink_rich_menus([user_id])
